test/unit/api/api-llm.test.js

```diff
@@ -121,27 +121,58 @@ tap.test('Agent API LLM methods', (t) => {
     })
   })
 
-  t.test('withLlmCustomAttributes', (t) => {
+  t.test('withLlmCustomAttributes should handle no active transaction', (t) => {
```

Comment:
Couldn't group tests here since helper requires a single instance of agent and would through if you create an inner test.

Reply:
Sorry, I don't understand this part: "...and would through if you create an inner test."

What is the issue?

Reply:
If you create a group of tests like below...

```js
t.test('some API test', () => {
  t.test('inner', () => {
      t.autoend()
      t.equal(1,1)
      // test would throw an error here due to how the suit is setup with mocking an agent if beforeEach / afterEach
      
  })
})
```

```
Object.loadMockedAgent (test/lib/agent_helper.js:105:22)
                Test.<anonymous> (test/unit/api/api-llm.test.js:29:26)
              at:
                line: 105
                column: 22
                file: test/lib/agent_helper.js
                function: Object.loadMockedAgent
              test: inner
              source: >2
                  _agent = new Agent(config)
                  _agent.__created = new Error('Only one agent at a time! This one was created at:')
                ---------------------^
                  _agent.recordSupportability = () => {} // Stub supportabilities.
```

Reply:
Please review the `beforeEach` and `afterEach` test scoped hooks. Yes, the agent is a singleton, but we have ways of re-bootstrapping it for each subtest.

Reply:
you don't have to group. what you did was fine, i'm going to tweak them slightly though